src/evaluators/ThrowStatement.js

```diff
@@ -24,5 +25,10 @@ export default function(
 ): Value {
   let exprRef = env.evaluate(ast.argument, strictCode);
   let exprValue = Environment.GetValue(realm, exprRef);
+  if (realm.isInPureScope() && !realm.isInPureTryStatement) {
+    invariant(realm.generator !== undefined);
+    realm.generator.emitThrow(exprValue);
```

Comment:
Technically, we'd want to materialize the `exprValue` before throwing it. Cc @sb98052 
However, same applies to the other 2 places where `emitThrow` is already used. Maybe just leave behind a comment for now.